pydra/engine/node.py

```diff
@@ -242,23 +243,34 @@ def cache_dir(self):
     @cache_dir.setter
     def cache_dir(self, location):
         if location is not None:
-            self._cache_dir = Path(location)
+            self._cache_dir = Path(location).resolve()
             self._cache_dir.mkdir(parents=False, exist_ok=True)
         else:
             self._cache_dir = mkdtemp()
             self._cache_dir = Path(self._cache_dir)
 
+    @property
+    def cache_locations(self):
+        return self._cache_locations
+
+    @cache_locations.setter
+    def cache_locations(self, locations):
+        if locations is not None:
+            self._cache_locations = [Path(loc) for loc in ensure_list(locations)]
+        else:
+            self._cache_locations = []
+
     @property
     def output_dir(self):
         return self._cache_dir / self.checksum
 
     def audit_check(self, flag):
         return self.audit_flags & flag
 
-    def __call__(self, cache_locations=None, **kwargs):
-        return self.run(cache_locations=cache_locations, **kwargs)
+    def __call__(self, **kwargs):
+        return self.run(**kwargs)
 
-    def run(self, cache_locations=None, **kwargs):
+    def run(self, **kwargs):
```

Comment:
this should check for `cache_locations` in `kwargs` so that people can update it in a call.

Reply:
ok, will work on this. As I said it would require more changes, so this updates is also sent to object that actually does the job

Reply:
it actually has to be only updated for `job`. Do you want to merge `cache_locations` from `NodeBase.__init__` and `Submitter.run` or overwrite? For now I'm overwriting:
https://github.com/satra/pydra/pull/10/files#diff-fecf4adafbe4fa0a6b80f1f04010f854R58